### PR TITLE
Henry flashcards

### LINKED ISSUE #46: Back-End Flashcards Storage Implementation
Description: Need to set up the flashcards sets to be held in the backend server and be associated to languages so each are toggled depending on this. This allows them to be saved and retrieved when needed.

Expected Outcome: Functional server connection for flashcards robust enough to store many flashcards for different languages as well as ones made elsewhere on the site (not made on flashcard page).

### PR DESCRIPTION
**Pull request by Henry-WK**
Added the backend of flashcards, can't say it works but it is something that can be shown and uses the architecture. There must be something small in it that im missing but I cant seem to find it!

### CODE REVIEW

File: backend/src/controllers/FlashcardController.js [hunk 1]
```diff
@@ -5,77 +5,141 @@ class FlashcardController {
...
+    //     const model = await this.modelPromise;
+    //     const Flashcard = model.getModel();
+    //     const deleted = await Flashcard.destroy({ where: { flashcardID: id } });
+    //     if (!deleted) return res.status(404).json({ success: false, error: "Flashcard not found." });
+    //     return res.status(200).json({ success: true, message: "Flashcard deleted." });
+    //   } catch (error) {
+    //     console.error("Error deleting flashcard:", error);
+    //     return res.status(500).json({ success: false, error: "Error deleting flashcard." });
+    //   }
+    // }
```
Comment (chrispho) [inline backend/src/controllers/FlashcardController.js:142]: Should these be commented out?

File: backend/src/models/SQLiteFlashcardModel.js [hunk 2]
```diff
@@ -16,81 +14,151 @@ class _SQLiteFlashcardModel {
...
+//       console.error("Error destroying flashcard:", error);
+//       throw new Error("Unable to destroy flashcard.");
+//     }
+//   }
+
+//   getModel() {
+//     return Flashcard;
+//   }
+// }
+
```
Comment (chrispho) [inline backend/src/models/SQLiteFlashcardModel.js:162]: Should this also be commented out?

File: backend/src/routes/flashcardRoutes.js [hunk 3]
```diff
@@ -8,40 +8,57 @@ class FlashcardRoutes {
...
+  //     console.log(`PUT /flashcards/${req.params.id}`);
+  //     await FlashcardController.updateFlashcard(req, res);
+  //   });
+
+  //   // Delete Flashcard
+  //   this.router.delete("/flashcards/:id", async (req, res) => {
+  //     console.log(`DELETE /flashcards/${req.params.id}`);
+  //     await FlashcardController.deleteFlashcard(req, res);
+  //   });
+  // }
```
Comment (chrispho) [inline backend/src/routes/flashcardRoutes.js:61]: same thing, purpose of commented code?

File: frontend/src/components/FlashcardsPageComponent/FlashcardsPageComponent.js [hunk 4]
```diff
@@ -293,77 +312,83 @@ export class FlashcardPageComponent extends Component {
...
+            //             addingText.style.display = 'none';
+
+            //             adding = false; //Reset adding info
+            //         }
+            //     }
+            //     else{
+            //         viewset.style.display = 'flex';
+            //         const setName = domSet.getAttribute("data-set-name");
+            //         currSet = setName;
+            //         this.#container.querySelector("#set-name").textContent = setName;
```
Comment (chrispho) [inline frontend/src/components/FlashcardsPageComponent/FlashcardsPageComponent.js:424]: more commented code, should this be commented out?

**Review by chrispho — CHANGES_REQUESTED** [review]
lots of commented code that can be deleted or explained

**Comment by Henry-WK** [comment]
I had alot commented because it was the original atchiecture from shiv. However, i will delete for clarity and because it is not going to be used ever at this point. THanks

**Comment by Henry-WK** [comment]
Going to redo a pull request with updates to my branch

**Pull request closed without merging**